mingw64 curl git→libtiff git cmake error

libtiff sure likes to break a lot.
```
Compiling the C compiler identification source file "CMakeCCompilerId.c" failed.
Compiler: C:/media-autobuild_suite/local64/bin/gcc.bat 
Build flags: -mthreads;-mtune=generic;-O2;-pipe
Id flags: --target=arm-arm-none-eabi;-mcpu=cortex-m3 

The output was:
1
gcc.exe: error: generic: No such file or directory
gcc.exe: error: arm-arm-none-eabi: No such file or directory
gcc.exe: error: cortex-m3: No such file or directory
gcc.exe: error: unrecognized command line option '-mtune'
gcc.exe: error: unrecognized command line option '-mcpu'; did you mean '-mcpu='?


Determining if the C compiler works failed with the following output:
Change Dir: C:/media-autobuild_suite/build/libtiff-git/build-64bit/CMakeFiles/CMakeTmp

Run Build Command(s):C:/media-autobuild_suite/msys64/mingw64/bin/ninja.exe cmTC_e81b8 && [1/2] Building C object CMakeFiles/cmTC_e81b8.dir/testCCompiler.c.obj

FAILED: CMakeFiles/cmTC_e81b8.dir/testCCompiler.c.obj 

C:\media-autobuild_suite\local64\bin\gcc.bat   -mthreads -mtune=generic -O2 -pipe -o CMakeFiles/cmTC_e81b8.dir/testCCompiler.c.obj   -c testCCompiler.c

gcc.exe: error: generic: No such file or directory
gcc.exe: error: unrecognized command line option '-mtune'
ninja: build stopped: subcommand failed.
```

[logs.zip](https://0x0.st/zE8V.zip)

--- a/build/media-suite_helper.sh
+++ b/build/media-suite_helper.sh
@@ -1965,39 +1965,18 @@ EOF
 }
 
 create_ab_ccache() {
-    local bin
+    local bin temp_file
+    temp_file=$(mktemp)
     for bin in {$MINGW_CHOST-,}{gcc,g++} clang{,++} cc cpp c++; do
-        if [[ ! -f "$LOCALDESTDIR/bin/$bin.bat" ]] || ! "$LOCALDESTDIR/bin/$bin.bat" --help > /dev/null 2>&1; then
-            printf '%s\r\n' \
-                '@echo off >nul 2>&1' \
-                'rem() { "$@"; }' \
-                'rem test -f nul && rm nul' \
-                'rem ccache > /dev/null 2>&1' \
-                "rem test \$? -ne 127 && ccache $bin \"\$@\"" \
-                'rem exit $?' \
-                "$(cygpath -m "$MINGW_PREFIX/bin/ccache") $bin %*" \
-                'exit %ERRORLEVEL%' \
-                'goto :EOF' \
-                ':args' \
-                'if -%1-==-- (' \
-                '    exit /b' \
-                ')' \
-                'test -f %1' \
-                'if %ERRORLEVEL%==0 (' \
-                '    for /f "tokens=1 delims=" %%a in ("cygpath -m %1") do set "ARGS=%ARGS% %%a"' \
-                '    shift' \
-                ') else (' \
-                '    test -d %1' \
-                '    if %ERRORLEVEL%==0 (' \
-                '        for /f "tokens=1 delims=" %%a in ("cygpath -m %1") do set "ARGS=%ARGS% %%a"' \
-                '        shift' \
-                '    ) else (' \
-                '        set "ARGS=%ARGS% %1"' \
-                '        shift' \
-                '    )' \
-                ')' \
-                'goto :args' > "$LOCALDESTDIR/bin/$bin.bat"
-        fi
+        cat << EOF > "$temp_file"
+@echo off >nul 2>&1
+rem() { "\$@"; }
+rem test -f nul && rm nul
+rem $(command -v ccache) --help > /dev/null 2>&1 && $(command -v ccache) $(command -v $bin) "\$@" || $(command -v $bin) "\$@"
+rem exit \$?
+$(cygpath -m "$(command -v ccache)") $(cygpath -m "$(command -v $bin)") %*
+EOF
+        diff -q "$temp_file" "$LOCALDESTDIR/bin/$bin.bat" > /dev/null || cp -f "$temp_file" "$LOCALDESTDIR/bin/$bin.bat"
         chmod +x "$LOCALDESTDIR/bin/$bin.bat"
     done
 }

